pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -18,7 +18,16 @@ in rec {
   #
   # * https://nixos.org/nix/manual/#ssec-derivation
   #   Explanation about derivations in general
-  mkDerivation =
+  mkDerivation = fnOrAttrs:
+    let
+      fn =
+        if builtins.isFunction fnOrAttrs
+        then fnOrAttrs
+        else _: fnOrAttrs;
+      this = mkDerivation_ (transform: mkDerivation (transform fn)) (fn { inherit this; });
```

Comment:
Oh, I see, it's just `this` being passed the first time. So we are all `this` / `super` today, with no `self`. Hmm.

Reply:
Sounds about right. This context is outdated though, so I'll resolve this conversation.